sycl/include/CL/__spirv/spirv_ops.hpp

```diff
@@ -155,6 +155,7 @@ extern SYCL_EXTERNAL TempRetT __spirv_ImageSampleExplicitLod(SampledType,
       macro(__attribute__((opencl_local)), Arg)
 
 __SPIRV_ATOMICS(__SPIRV_ATOMIC_FLOAT, float)
+__SPIRV_ATOMICS(__SPIRV_ATOMIC_FLOAT, double)
```

Comment:
Just a question: Is 'half' type expected to be supported too eventually?

Reply:
Good question.  OpenCL only defines atomics for 32- and 64-bit types, which is what I've started with here.

C++ allows an `atomic_ref` to be constructed from any `TriviallyCopyable` type, and I think eventually that's where we'd like to end up.  Whether the `atomic_ref` for a given type is implemented on top of native instructions, a compare-exchange loop, a global lock, etc, would then be implementation-defined and device-specific.

We haven't yet defined exactly what sort of device queries we'll want to support for testing atomic functionality.  If you have any ideas for what the feature sets should look like, or if there are any queries that would be useful for implementing reductions, please let me know!